pydoctor/epydoc2stan.py

```diff
@@ -850,8 +849,9 @@ def visit_Subscript(self, node: ast.Subscript) -> Tag:
         sub: ast.AST = node.slice
         if isinstance(sub, ast.Index):
             # In Python < 3.9, non-slices are always wrapped in an Index node.
-            if sys.version_info < (3, 9):
-                sub = sub.value
+            # If tox runs with python 3.9, mypy will get a warning. 
+            # But we can't reach this code with python 3.9 anyway.
+            sub = sub.value # type: ignore[attr-defined]
```

Comment:
mypy is supposed to [understand `sys.version_info`](https://mypy.readthedocs.io/en/stable/common_issues.html#version-and-platform-checks), so in theory this shouldn't be a problem.

Reply:
Well.. I get a warning on python 3.9...

Reply:
So you'r saying that we should add an `assert sys.version_info < (3, 9)` check ?

Reply:
I can't reproduce the problem under Python 3.9 if the code looks like this:
```py
        if isinstance(sub, ast.Index):
            if sys.version_info < (3, 9):
                # In Python < 3.9, non-slices are always wrapped in an Index node.
                sub = sub.value
```
Only if I remove the `sys.version_info < (3, 9)` check, I see the warning.

To test, I added `basepython = python3.9` under `[testenv:mypy]` in `tox.ini` and added `python --version` to the command; it prints 3.9.5.

As for a solution, I think this is probably the safest approach:
```py
        if sys.version_info < (3, 9) and isinstance(sub, ast.Index):
            # In Python < 3.9, non-slices are always wrapped in an Index node.
            sub = sub.value
```
This will still work if the `Index` class is removed at some point.

Reply:
Done!